### PR TITLE
Highlight selected events in the query builder

### PR DESCRIPTION
**Pull request by ranquild**
Epic: https://github.com/metabase/metabase/issues/20289

Changes:
- Clicking on the event icon in the chart highlights the event (group) and opens the sidebar
- Clicking again deselects the event (group)
- All timelines are available in the sidebar in the query builder
- Timelines within the question's collection are shown by default. As a result, timelines within `Our analytics` are shown for unsaved questions by default. We might want to fix this later.
- A timeline is expanded if it's the only available timeline or if there are highlighted events

How to test:
- Go to a question
- Click on the calendar icon in the bottom right corner and create events, if there are none
- Make sure they are displayed on the chart
- Click on the event and make sure it's highlighted and the sidebar is opened

Screenshot 2022-03-17 at 18 42 48

### CODE REVIEW

File: frontend/src/metabase/query_builder/containers/QueryBuilder.jsx [hunk 1]
```diff
@@ -326,6 +337,7 @@ function QueryBuilder(props) {
 
 export default _.compose(
   Bookmark.loadList(),
+  Timelines.loadList(timelineProps),
```
Comment (kulyk) [inline frontend/src/metabase/query_builder/containers/QueryBuilder.jsx:340]: Are we sure we want to block the QB rendering until the timelines are fetched?

File: frontend/src/metabase/query_builder/reducers.js [hunk 2]
```diff
@@ -78,7 +81,7 @@ const DEFAULT_UI_CONTROLS = {
   queryBuilderMode: false, // "view" | "notebook" | "dataset"
   previousQueryBuilderMode: false,
   snippetCollectionId: null,
-  datasetEditorTab: "query", // "query" / "metadata"
+  datasetEditorTab: "query", // "query" / "metadata",
```
Comment (kulyk) [inline frontend/src/metabase/query_builder/reducers.js:84]: Why is there a comma here?

File: frontend/src/metabase/query_builder/reducers.js [hunk 3]
```diff
@@ -500,16 +503,40 @@ export const currentState = handleActions(
   null,
 );
 
-export const timelineVisibility = handleActions(
+export const timelineIds = handleActions(
```
Comment (kulyk) [inline frontend/src/metabase/query_builder/reducers.js:506]: nip: wonder if it'd be better to call it something like `visibleTImelineIds`? Just to make it clear this state handles timelines visibility on charts and nothing else

**Review by kulyk — APPROVED** [review]
Works like a charm! Nice work

File: frontend/src/metabase/query_builder/containers/QueryBuilder.jsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/query_builder/containers/QueryBuilder.jsx:340]: There is `loadingAndErrorWrapper: false`, it's not blocking

File: frontend/src/metabase/query_builder/reducers.js (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/query_builder/reducers.js:84]: Not intended, fixing

File: frontend/src/metabase/query_builder/reducers.js (hunk 3 above)
Comment (ranquild) [reply frontend/src/metabase/query_builder/reducers.js]: Renamed this to `visibleTimelineIds` and `timelineEventIds` to `selectedTimelineIds`. Actions are renamed accordingly. Thanks!

**Pull request merged**